bin/s3fire_downloader.py

```diff
@@ -0,0 +1,141 @@
+    """
+    Downloads an individual file from FIRE S3 using its object key.
+
+    :param s3_key: The S3 object key of the file to download.
+    :type s3_key: str
+    :param bucket: The name of the S3 bucket.
+    :type bucket: str
+    :param outdir: The local directory to save the downloaded file.
+    :type outdir: str
+    :param access_key: The access key for private S3 buckets (optional for public files).
+    :type access_key: Optional[str]
+    :param secret_key: The secret key for private S3 buckets (optional for public files).
+    :type secret_key: Optional[str]
+    :return: None
+    :rtype: None
+    :raises ValueError: If credentials are missing for private files.
+    :raises Exception: For other download errors.
+    """
+    s3_args = {"endpoint_url": FIRE_ENDPOINT}
+    if bucket == PRIVATE_BUCKET:
+        if not access_key or not secret_key:
+            logger.error("Missing credentials for private files.")
+            raise ValueError("Access key and secret key are required for private files.")
+        s3_args.update(
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+        )
+    else:
+        # Public bucket configuration with unsigned requests
+        s3_args.update({"config": Config(signature_version=UNSIGNED)})
+
+    s3 = boto3.client("s3", **s3_args)
+
+    os.makedirs(outdir, exist_ok=True)
+    local_file_path = os.path.join(outdir, os.path.basename(s3_key))
+
+    try:
+        logger.info(f"Downloading {s3_key} from S3 bucket {bucket} to {local_file_path}...")
+        s3.download_file(bucket, s3_key, local_file_path)
+        logger.info(f"File successfully downloaded to: {local_file_path}")
+    except Exception as e:
+        logger.error(f"Error downloading file from S3: {e}")
+        raise
+
+
+def download_files(ftp_paths: List[str], outdir: str, access_key: Optional[str], secret_key: Optional[str]) -> None:
+    """
+    Downloads multiple files from their FTP paths.
+
+    :param ftp_paths: List of FTP paths to download.
+    :type ftp_paths: List[str]
+    :param outdir: Directory to save the downloaded files.
+    :type outdir: str
+    :param access_key: Access key for private files.
+    :type access_key: Optional[str]
+    :param secret_key: Secret key for private files.
+    :type secret_key: Optional[str]
+    """
+    for ftp_path in ftp_paths:
+        try:
+            s3_key, bucket = transform_ftp_to_s3(ftp_path)
+            download_file_from_fire(s3_key, bucket, outdir, access_key, secret_key)
+        except ValueError as ve:
+            logger.error(f"Skipping download due to error: {ve}")
+        except Exception as e:
+            logger.error(f"Unexpected error while downloading {ftp_path}: {e}")
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Download multiple files from FTP paths via FIRE S3 (supports public and private files)."
+    )
+    parser.add_argument(
+        "--ftp_paths",
```

Comment:
```suggestion
        "--ftp-paths",
```
just to match the format of other args.